javasphinx/apidoc.py

```diff
@@ -192,8 +200,14 @@ def generate_from_source_file(doc_compiler, source_file, cache_dir):
     except Exception:
         util.unexpected('Unexpected exception while parsing %s', source_file)
 
+    documents = {}
     try:
-        documents = doc_compiler.compile(ast)
+        if source_file.endswith("package-info.java"):
+            for _, node in ast.filter(javalang.tree.PackageDeclaration):
```

Comment:
I think a filter here is overkill. You should just be able to check that `ast.package` is not `None`.